src/pages/NotePage/Notes.jsx

```diff
@@ -1,13 +1,86 @@
 import "./Notes.css"
 import { NavBar, Aside } from "../../components"
 import { useState } from "react";
-
+import { useData } from "../../context/data-context";
+import { useAuth } from "../../context/auth-context";
+import { postNote,editNote,deleteNote } from "../../services/Services";
+import { NoteCard } from "../../components";
 
 export const Notes=()=>{
 const [noteCard,setNoteCard] = useState(false);
+
 const expandCard=()=>{
     setNoteCard(true);
 }
+const date = new Date();
+const initialVal= {
+    _id:"",
+    title:"",
+    body:"",
+    backGround:"#e7dcdc",
+    CreatedAt:`${date.getDate()}/0${
+        date.getMonth() + 1
+      }/${date.getFullYear()}`,
+}
+
+const [note,setNote]= useState(initialVal);
+const {noteState,noteDispatch}= useData();
+
+const notesArray= noteState.noteData;
+
+const {token}= useAuth();
+const addNoteHandler=async()=>{
+   
+   try{
+       if(!note._id)
+       {
+        const response= await postNote({token,note});
+        
+        if(response.status===200 || response.status===201)
```

Comment:
The if conditions and the else condition are the same except the postNote and editNote parts. so you could extract out the common functionality into a separate function.